tests/test_user_update.py

```diff
@@ -0,0 +1,258 @@
+    payload_dict = {
+        "identifier": {
+            "user_id": user_id
+        },
+        "first_name": "stringy",
+        "email": "user@example456.com",
+        "password": "string2"
+    }
+
+    return payload_dict
+
+
+def get_update_request_no_data() -> Dict[str, Any]:
+    """
+    Generates an invalid request that contains no information
+    """
+    payload_dict = {}
+    return payload_dict
+
+
+def check_response_valid_update(response: HTTPResponse) -> bool:
+    """
+    Checks if response code to successful user update is valid
+    """
+    return response.status_code == 200
+
+
+def check_response_invalid_fields(response: HTTPResponse) -> bool:
+    """
+    Checks if response code to user update with illegal fields is valid
+    """
+    return response.status_code == 422
+
+
+def check_response_update_nonexistent(response: HTTPResponse) -> bool:  # pylint: disable=invalid-name
+    """
+    Checks if response code to user update with incorrect identifier is valid
+    """
+    return response.status_code == 404
+
+
+def check_response_no_data(response: HTTPResponse) -> bool:
+    """
+    Checks if response code to user update with no data provided is valid
+    """
+    return response.status_code == 422
+
+
+def check_fields_updated_correctly(
+        old_user_data: user_models.User,
+        updated_data_json: Dict[str, Any]) -> bool:
+    """
+    Checks the updated user response VS the original user data
+    and returns True if the operation outcome was valid, else False.
+
+    todo: refactor: loop through new user fields.
+    If field is in updated_data_json, expect that. If not, expect
+    it to be whatever it is in the old user model.
+    """
+
+    new_user_data = get_user_data_from_user_model(old_user_data)
+    unchecked_items = old_user_data.dict()
+
+    # todo: check last name field WASN'T changed
+    for key, val in updated_data_json.items():
+        # fixme: complex code, don't like
+        if new_user_data.dict().get(key) != val \
+                and key not in get_attributes_to_be_different():
+            if key in get_attributes_to_be_hashed():
+                if not new_user_data.check_password(val):
+                    return False
+            else:
+                return False
+        # trims down a list to values that SHOULDN'T have been updated
+        if key in unchecked_items:
+            del unchecked_items[key]
+
+    for key, val in unchecked_items.items():
+        if new_user_data.dict().get(key) != val:
+            return False
+    return True
+
+
+def check_fields_not_updated(old_user_data: user_models.User) -> bool:
+    """
+    Checks fields in Mongo are the same as they are in the provided user data.
+    """
+    new_user_data = get_user_data_from_user_model(old_user_data)
+
+    for key, val in old_user_data.dict().items():
+        if key in get_attributes_to_be_hashed():
+            if not new_user_data.check_password(val):
+                return False
```

Comment:
I appreciate the intent behind `get_attributes_to_be_hashed()` for making this easier to extend in the future, but we get a hint that this is unnecessary because of the next call to `user.check_password`. 

This probably tells us the we can make this into a single method that _only_ deals with passwords, which will get rid of this nested `if` nightmare.